src/otx/algo/instance_segmentation/losses/rpn_loss.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) OpenMMLab. All rights reserved.
+#
+"""ATSS criterion."""
+
+from __future__ import annotations
+
+from torch import Tensor, nn
+
+from otx.algo.common.utils.utils import multi_apply
+
+
+class RPNCriterion(nn.Module):
+    """RPNCriterion is a loss criterion used in the Region Proposal Network (RPN) algorithm.
+
+    Args:
+        num_classes (int): The number of object classes.
+        bbox_coder (nn.Module): The module used for encoding and decoding bounding box coordinates.
+        loss_cls (nn.Module): The module used for calculating the classification loss.
+        loss_bbox (nn.Module): The module used for calculating the bounding box regression loss.
+        loss_centerness (nn.Module | None, optional): The module used for calculating the centerness loss.
+            Defaults to None.
+        use_qfl (bool, optional): Whether to use the Quality Focal Loss (QFL).
+            Defaults to ``CrossEntropyLoss(use_sigmoid=True, loss_weight=1.0)``.
+        reg_decoded_bbox (bool, optional): Whether to use the decoded bounding box coordinates
+            for regression loss calculation. Defaults to True.
+        bg_loss_weight (float, optional): The weight for the background loss.
+            Defaults to -1.0.
+    """
+
+    def __init__(
+        self,
+        bbox_coder: nn.Module,
+        loss_cls: nn.Module,
+        loss_bbox: nn.Module,
+    ) -> None:
+        super().__init__()
+        self.bbox_coder = bbox_coder
+        self.loss_bbox = loss_bbox
+        self.loss_cls = loss_cls
+        self.cls_out_channels = 1 if loss_cls.use_sigmoid else 2
+
+    def forward(
+        self,
+        cls_reg_targets: tuple[list[Tensor], list[Tensor], list[Tensor], list[Tensor], int],
+        bbox_preds: list[Tensor],
+        cls_scores: list[Tensor],
+    ) -> dict:
+        """Calculate the loss based on the features extracted by the detection head.
+
+        TODO (kirill): it is related to RPNHead for iseg, will be deprecated
```

Comment:
Should this be removed as well?

Reply:
Done https://github.com/openvinotoolkit/training_extensions/pull/3865/commits/0f4409c90de69578b2e4b09e66f2b5b4c072e436